src/test/kotlin/org/rust/ide/inspections/import/AutoImportFixTest.kt

```diff
@@ -1598,4 +1604,160 @@ class AutoImportFixTest : AutoImportFixTestBase() {
+        }
+
+        fn main() {
+            let f = Bar/*caret*/;
+        }
+    """)
+
+    fun `test add import to existing group`() = checkAutoImportFixByText("""
+        use foo::{Foo, Bar};
+
+        mod foo {
+            pub struct Foo;
+            pub struct Bar;
+            pub struct Baz;
+        }
+
+        fn main() {
+            let f = <error descr="Unresolved reference: `Baz`">Baz/*caret*/</error>;
+        }
+    """, """
+        use foo::{Foo, Bar, Baz};
+
+        mod foo {
+            pub struct Foo;
+            pub struct Bar;
+            pub struct Baz;
+        }
+
+        fn main() {
+            let f = Baz/*caret*/;
+        }
+    """)
+
+    fun `test group imports only at the last level`() = checkAutoImportFixByText("""
+        use foo::Foo;
+
+        mod foo {
+            pub struct Foo;
+            pub mod bar {
+                pub struct Bar;
+            }
+        }
+
+        fn main() {
+            let f = <error descr="Unresolved reference: `Bar`">Bar/*caret*/</error>;
+        }
+    """, """
+        use foo::Foo;
+        use foo::bar::Bar;
+
+        mod foo {
+            pub struct Foo;
+            pub mod bar {
+                pub struct Bar;
+            }
+        }
+
+        fn main() {
+            let f = Bar/*caret*/;
+        }
+    """)
+
+    fun `test group imports only if the other import doesn't have a modifier`() = checkAutoImportFixByText("""
+        pub use foo::Foo;
+
+        mod foo {
+            pub struct Foo;
+            pub struct Bar;
+        }
+
+        fn main() {
+            let f = <error descr="Unresolved reference: `Bar`">Bar/*caret*/</error>;
+        }
+    """, """
+        pub use foo::Foo;
+        use foo::Bar;
+
+        mod foo {
+            pub struct Foo;
+            pub struct Bar;
+        }
+
+        fn main() {
+            let f = Bar/*caret*/;
+        }
+    """)
+
+    fun `test group imports only if the other import doesn't have an attribute`() = checkAutoImportFixByText("""
+        #[attribute = "value"]
+        use foo::Foo;
+
+        mod foo {
+            pub struct Foo;
+            pub struct Bar;
+        }
+
+        fn main() {
+            let f = <error descr="Unresolved reference: `Bar`">Bar/*caret*/</error>;
+        }
+    """, """
+        #[attribute = "value"]
+        use foo::Foo;
+        use foo::Bar;
+
+        mod foo {
+            pub struct Foo;
+            pub struct Bar;
+        }
+
+        fn main() {
+            let f = Bar/*caret*/;
+        }
+    """)
+
+    fun `test group imports only if the other import doesn't have an alias`() = checkAutoImportFixByText("""
+        use foo::Foo as F;
+
+        mod foo {
+            pub struct Foo;
+            pub struct Bar;
+        }
+
+        fn main() {
+            let f = <error descr="Unresolved reference: `Bar`">Bar/*caret*/</error>;
+        }
+    """, """
+        use foo::Foo as F;
+        use foo::Bar;
```

Comment:
Are you sure that we don't want to merge use items in this case?
`use foo::{Foo as F, Bar}` is valid and semantically the same

Reply:
Fixed.